recipes/lora_finetune_single_device.py

```diff
@@ -156,20 +156,18 @@ def _update_recipe_state(self, ckpt_dict: Dict[str, Any]) -> None:
         """
         Updates the recipe state from checkpoint.
         """
-        # If seed, total_epoch or max_steps_per_epoch don't match,
+        # If seed or max_steps_per_epoch don't match,
         # warn the user and overwrite
         if (
             self.seed != ckpt_dict[utils.SEED_KEY]
-            or self.total_epochs != ckpt_dict[utils.TOTAL_EPOCHS_KEY]
             or self.max_steps_per_epoch != ckpt_dict[utils.MAX_STEPS_KEY]
         ):
             warn(
-                message="""Configured value for seed, epochs or max_steps_per_epoch
+                message="""Configured value for seed or max_steps_per_epoch
                 does not match the value stored in checkpoint."""
             )
         self.seed = utils.set_seed(seed=ckpt_dict[utils.SEED_KEY])
         self.epochs_run = ckpt_dict[utils.EPOCHS_KEY]
-        self.total_epochs = ckpt_dict[utils.TOTAL_EPOCHS_KEY]
```

Comment:
@joecummings I made a quick fix to the `test_training_state_on_resume` test so that we'll notice errors like this, but it required slightly different configurations for the pre- and post-resume runs (first `epochs=2` and then `epochs=3`), which was not allowed by this `_update_recipe_state` above, so i removed the overwrite for total epochs. 

Is there a reason for this overwrite? IMO we should allow users to resume a training job with a different number of total epochs, like for example if they finish a training job and then decide they want to continue training for a few more epochs. To support this, we should also save the recipe state during the final checkpoint.

Thoughts? Should I remove the epoch overwrite for the other recipes? Or should I find a different way to test for this error?

Reply:
I 100% agree with this approach, but maybe we should opt to be overly communicative here? Log a warning if the total epochs from the ckpt_dict differ from the total epochs requested?